xmtp_mls/src/groups/mod.rs

```diff
@@ -614,6 +615,13 @@ where
             let disappearing_settings = mutable_metadata.clone().and_then(|metadata| {
                 Self::conversation_message_disappearing_settings_from_extensions(metadata).ok()
             });
+
+            if let Some(group) = provider.db().find_group(group_id.as_slice())?{
+                if group.membership_state == GroupMembershipState::Allowed {
```

Comment:
Are we sure the `GroupMembershipState` is the thing we want to be using to make this decision? If we have moved the group to "rejected" or the membership state is still "pending", from an MLS perspective nothing is different.

I worry this is going to be a footgun.